app/src/context/Mnemonic.tsx

```diff
@@ -0,0 +1,76 @@
+import * as React from "react";
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import "react-native-get-random-values";
+import "@ethersproject/shims";
+import { ethers } from "ethers";
+import AsyncStorage from "@react-native-async-storage/async-storage";
+import { mnemonicLocalStorage } from "../utils/local-storage";
+
+export type MnemonicValue = {
+  mnemonic: string | undefined;
+  createMnemonic: () => Promise<string | undefined>;
+  reset: () => void;
+};
+
+export const MnemonicContext = React.createContext<MnemonicValue | undefined>(
+  undefined
+);
+
+export const MnemonicProvider: React.FC<{
+  children: React.ReactNode;
+}> = (props) => {
+  const [mnemonic, setMnemonic] = React.useState<string | undefined>();
+
+  React.useEffect(() => {
+    (async () => {
+      const initialMnemonic = await mnemonicLocalStorage.get();
+
+      if (initialMnemonic) {
+        setMnemonic(initialMnemonic.privateKey);
+      }
+    })();
+  }, []);
+
+  const createMnemonic = async () => {
+    try {
+      const wallet = ethers.Wallet.createRandom();
+      await AsyncStorage.setItem("PRIVATE_KEY", wallet.privateKey);
+      setMnemonic(wallet.mnemonic.phrase);
+      console.log(mnemonic)
+      return mnemonic;
+    } catch (error) {
+      console.log(error);
```

Comment:
What about this one also @jrocca82

Reply:
No, need to catch for error, this is standard for a try/catch function